packages/remix-dev/config/routesConvention.ts

```diff
@@ -49,6 +49,10 @@ export function defineConventionalRoutes(
   });
 
   let routeIds = Object.keys(files).sort(byLongestFirst);
+  let parentRouteIds: { [routeId: string]: string | undefined } = {};
+  for (let id of routeIds) {
+    parentRouteIds[id] = findParentRouteId(routeIds, id);
```

Comment:
This is the only usage of `findParentRouteId` in the codebase, so it would be good to inline it here:

```suggestion
    parentRouteIds[id] = routeIds.find((id) => childRouteId.startsWith(`${id}/`));
```

Optionally, you could then factor out the `parentRouteIds` calculation into a `getParentRouteIds(routeIds)` function.

So this whole block would become:

```ts
let parentRouteIds = getParentRouteIds(routeIds)
```

Reply:
Done, I pulled the whole thing into a separate function.